src/Aspire.Hosting.Maui/MauiMacCatalystExtensions.cs

```diff
@@ -0,0 +1,120 @@
+    /// <para>
+    /// The resource name will default to "{projectName}-maccatalyst".
+    /// </para>
+    /// </remarks>
+    /// <example>
+    /// Add a Mac Catalyst device to a MAUI project:
+    /// <code lang="csharp">
+    /// var builder = DistributedApplication.CreateBuilder(args);
+    /// 
+    /// var maui = builder.AddMauiProject("mauiapp", "../MyMauiApp/MyMauiApp.csproj");
+    /// var macCatalystDevice = maui.AddMacCatalystDevice();
+    /// 
+    /// builder.Build().Run();
+    /// </code>
+    /// </example>
+    public static IResourceBuilder<MauiMacCatalystPlatformResource> AddMacCatalystDevice(
+        this IResourceBuilder<MauiProjectResource> builder)
+    {
+        ArgumentNullException.ThrowIfNull(builder);
+
+        var name = $"{builder.Resource.Name}-maccatalyst";
+        return builder.AddMacCatalystDevice(name);
+    }
+
+    /// <summary>
+    /// Adds a Mac Catalyst device resource to run the MAUI application on the macOS platform with a specific name.
+    /// </summary>
+    /// <param name="builder">The MAUI project resource builder.</param>
+    /// <param name="name">The name of the Mac Catalyst device resource.</param>
+    /// <returns>A reference to the <see cref="IResourceBuilder{T}"/>.</returns>
+    /// <remarks>
+    /// This method creates a new Mac Catalyst platform resource that will run the MAUI application
+    /// targeting the Mac Catalyst platform using <c>dotnet run</c>. The resource does not auto-start 
+    /// and must be explicitly started from the dashboard by clicking the start button.
+    /// <para>
+    /// Multiple Mac Catalyst device resources can be added to the same MAUI project if needed, each with
+    /// a unique name.
+    /// </para>
+    /// </remarks>
+    /// <example>
+    /// Add multiple Mac Catalyst devices to a MAUI project:
+    /// <code lang="csharp">
+    /// var builder = DistributedApplication.CreateBuilder(args);
+    /// 
+    /// var maui = builder.AddMauiProject("mauiapp", "../MyMauiApp/MyMauiApp.csproj");
+    /// var macCatalystDevice1 = maui.AddMacCatalystDevice("maccatalyst-device-1");
+    /// var macCatalystDevice2 = maui.AddMacCatalystDevice("maccatalyst-device-2");
+    /// 
+    /// builder.Build().Run();
+    /// </code>
+    /// </example>
+    public static IResourceBuilder<MauiMacCatalystPlatformResource> AddMacCatalystDevice(
+        this IResourceBuilder<MauiProjectResource> builder,
+        [ResourceName] string name)
+    {
+        ArgumentNullException.ThrowIfNull(builder);
+        ArgumentException.ThrowIfNullOrWhiteSpace(name);
+
+        // Check if a Mac Catalyst device with this name already exists in the application model
+        var existingMacCatalystDevices = builder.ApplicationBuilder.Resources
+            .OfType<MauiMacCatalystPlatformResource>()
+            .FirstOrDefault(r => r.Parent == builder.Resource && 
+                                 string.Equals(r.Name, name, StringComparisons.ResourceName));
```

Comment:
You don't actually need to two this check (in fact the check is not quite right). You can just add the resource to the model and it will do a uniqueness check on the name. You cannot have two resources with the same name even if they differ by type. Change this for the Windows resource too - I didn't pick it up in review last time ;)